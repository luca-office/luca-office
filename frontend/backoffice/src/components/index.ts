export * from "./binary-handling"
export * from "./card"
export * from "./card-overview/card-overview"
export * from "./charts"
export * from "./creation-bar/creation-bar"
export * from "./creation-status/creation-status"
export * from "./editable-multiline-textarea/editable-multiline-textarea"
export * from "./editing-status-indicator/editing-status-indicator"
export * from "./email"
export * from "./information-entry/information-entry"
export * from "./inline-editable-header/inline-editable-header"
export * from "./inline-editable-textarea/inline-editable-textarea"
export * from "./markdown-text/markdown-text"
export * from "./markdown-text-field/markdown-text-field"
export * from "./markdown-text-update-modal/markdown-text-update-modal"
export * from "./meta-entry/meta-entry"
export * from "./meta-entry-tags/meta-entry-tags"
export * from "./overlay-edit-field/overlay-edit-field"
export * from "./rating-progress-card/rating-progress-card"
export * from "./rating-progress-card/utils"
export * from "./resort-modal/resort-modal"
export * from "./sub-header-detail/sub-header-detail"
export * from "./sub-header-detail/sub-header-detail-container"
export * from "./sub-header-filter/sub-header-filter"
export * from "./table-action-bar/table-action-bar"
export * from "./upload-file-modal/upload-file-modal"
