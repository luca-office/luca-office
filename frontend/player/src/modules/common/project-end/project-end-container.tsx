import * as React from "react"
import {useGetSurveyInvitationFromRedux} from "../../../hooks/use-get-survey-invitation"
import {ProjectEnd} from "./project-end"

export const ProjectEndContainer: React.FC = () => {
  const {tokenOption} = useGetSurveyInvitationFromRedux()
  return tokenOption.map(token => <ProjectEnd token={token} />).orNull()
}
