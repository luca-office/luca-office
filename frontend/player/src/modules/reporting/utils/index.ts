export * from "./participant-project-progress"
