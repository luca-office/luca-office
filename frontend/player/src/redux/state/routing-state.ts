import {RouteParameters, Route} from "../../routes"

export interface RoutingState {
  readonly activeRoute: Route
  readonly parameters?: RouteParameters
}

export const initialRoutingState = (surveyId: UUID | null): RoutingState => ({
  activeRoute: surveyId !== null ? Route.OpenParticipation : Route.VerifyToken,
  parameters: surveyId !== null ? {surveyId} : undefined
})
