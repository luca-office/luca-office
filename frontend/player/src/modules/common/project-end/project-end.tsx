import {css} from "@emotion/react"
import * as React from "react"
import {AppHeader, DesktopBackgroundOverlay, Heading, Text} from "shared/components"
import {styles as desktopStyles} from "shared/components/desktop/desktop.styles"
import {HeadingLevel} from "shared/enums"
import {Flex, FontWeight, headerHeight, spacingCard, spacingLarge, spacingSmall, TextSize} from "shared/styles"
import {useLucaTranslation} from "shared/translations"

interface Props {
  readonly token: string
}

export const ProjectEnd: React.FC<Props> = ({token}) => {
  const {t} = useLucaTranslation()
  return (
    <div css={desktopStyles.container}>
      <AppHeader title={t("project__end_successful_header")} />
      <DesktopBackgroundOverlay
        customStyles={styles.backgroundOverlay}
        customContentStyles={[styles.backgroundOverlayContent, styles.content]}>
        <div className="desktop-final-content" css={styles.content}>
          <Heading customStyles={{marginBottom: spacingCard}} level={HeadingLevel.h2}>
            {t("project__end_successful_title")}
          </Heading>
          <Text size={TextSize.Smaller}>{t("project__end_successful_text")}</Text>
          <div css={styles.tokenRow}>
            <Heading fontWeight={FontWeight.Bold} level={HeadingLevel.h3}>
              {t("project__start_dialog_text_confirm_modal_code")}
            </Heading>
            <Text customStyles={styles.token}>{token}</Text>
          </div>
        </div>
      </DesktopBackgroundOverlay>
    </div>
  )
}

const styles = {
  content: css(Flex.column, {
    alignItems: "center",
    justifyContent: "center"
  }),
  tokenRow: css(Flex.row, {
    marginTop: spacingLarge,
    alignItems: "center"
  }),
  token: css({
    marginLeft: spacingSmall,
    fontSize: TextSize.Medium
  }),
  backgroundOverlay: css({
    height: `calc(100vh - ${headerHeight}px)`
  }),
  backgroundOverlayContent: css({
    maxWidth: 600,
    margin: "auto"
  })
}
