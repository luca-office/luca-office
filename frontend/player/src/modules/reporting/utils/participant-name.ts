import {SurveyInvitation} from "shared/models"
import {Option} from "shared/utils"

export const getParticipantName = (surveyInvitationOption: Option<SurveyInvitation>) => {
  return surveyInvitationOption.map(surveyInvitation =>
    surveyInvitation?.participantData !== null && surveyInvitation?.participantData !== undefined
      ? `${surveyInvitation.participantData.firstName} ${surveyInvitation.participantData.lastName}`
      : surveyInvitation?.token
  ) as Option<string>
}
