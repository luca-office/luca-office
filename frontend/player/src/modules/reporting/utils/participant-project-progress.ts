import {compact} from "lodash-es"
import {Salutation} from "shared/graphql/generated/globalTypes"
import {ParticipantProjectProgress, ProjectModule, ProjectModuleScore, SurveyInvitation} from "shared/models"
import {find, getModuleProgress, isEmpty, Option} from "shared/utils"

interface GetParticipantProjectProgress {
  readonly surveyInvitation: SurveyInvitation
  readonly participantName: Option<string>
  readonly projectModules: ProjectModule[]
  readonly projectModulesScores: ProjectModuleScore[]
}

export const getParticipantProjectProgress = ({
  surveyInvitation,
  participantName,
  projectModules,
  projectModulesScores
}: GetParticipantProjectProgress): ParticipantProjectProgress => ({
  id: surveyInvitation.id,
  displayName: participantName.getOrElse(""),
  moduleProgress: compact(
    surveyInvitation.projectModuleProgresses
      .slice()
      .sort((a, b) => {
        const aPosition = find(
          ({scenarioId, questionnaireId}) =>
            a.scenarioId !== null ? scenarioId === a.scenarioId : questionnaireId === a.questionnaireId,
          projectModules
        )
          .map(module => module.position)
          .getOrElse(-1)

        const bPosition = find(
          ({scenarioId, questionnaireId}) =>
            b.scenarioId !== null ? scenarioId === b.scenarioId : questionnaireId === b.questionnaireId,
          projectModules
        )
          .map(module => module.position)
          .getOrElse(-1)

        return aPosition - bPosition
      })
      .map(projectModuleProgress => {
        const projectModule = projectModules.find(projectModule =>
          projectModuleProgress.scenarioId !== null
            ? projectModule.scenarioId === projectModuleProgress.scenarioId
            : projectModuleProgress.questionnaireId === projectModule.questionnaireId
        )

        if (projectModule !== undefined) {
          return getModuleProgress({
            projectModule,
            projectModuleProgresses: surveyInvitation.projectModuleProgresses,
            projectModulesScores
          })
        } else {
          if(!isEmpty(projectModules)) {
            // log error if project modules are loaded, but no related progress found
            console.warn("ProjectModule could not be found")
          }
          return null
        }
      })
  ),
  salutation: surveyInvitation.participantData?.salutation ?? Salutation.NonBinary,
  token: surveyInvitation.token
})
