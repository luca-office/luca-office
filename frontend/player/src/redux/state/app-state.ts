import {initialSharedAppState, SharedAppState} from "shared/redux/state/app-state"
import {initialRoutingState, RoutingState} from "./routing-state"

export type AppState = Readonly<{routing: RoutingState}> & SharedAppState

export const initialAppState = (surveyId: UUID | null): AppState => ({
  routing: initialRoutingState(surveyId),
  ...initialSharedAppState
})
