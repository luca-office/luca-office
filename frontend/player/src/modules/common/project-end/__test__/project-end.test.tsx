import {shallow} from "enzyme"
import * as React from "react"
import {create} from "react-test-renderer"
import {AppHeader, Heading, Text} from "shared/components"
import {getDateValueMock} from "sharedTests/utils/date-mock"
import {ShallowWrapper} from "sharedTests/utils/shallow-wrapper"
import {initialAppState} from "../../../../redux/state/app-state"
import {ProjectEnd} from "../project-end"

describe("ProjectEnd", () => {
  const _Date = Date
  beforeEach(() => {
    const mockDate = getDateValueMock(1970, 1, 1)
    Date = mockDate
  })
  afterEach(() => {
    Date = _Date
  })
  it("renders correctly", () => {
    const component = create(
      <ShallowWrapper initialAppState={initialAppState(null)}>{<ProjectEnd token="fsdskdjf" />}</ShallowWrapper>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("has correct structure", () => {
    const component = shallow(<ProjectEnd token="fsdskdjf" />, {wrappingComponent: ShallowWrapper})
    expect(component.find(AppHeader)).toHaveLength(1)
    expect(component.find(".desktop-final-content")).toHaveLength(1)
    expect(component.find(Heading)).toHaveLength(2)
    expect(component.find(Text)).toHaveLength(2)
  })
})
