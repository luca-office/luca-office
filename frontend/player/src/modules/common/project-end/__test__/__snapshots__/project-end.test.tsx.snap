// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectEnd renders correctly 1`] = `
<div
  className="css-hbphg0-ProjectEnd"
>
  <div
    className="css-1fm4cyn-AppHeader"
  >
    <div
      className="app-title css-5p7lab-AppHeader"
    >
      <div
        className="css-1fxkx3u-Icon"
        dangerouslySetInnerHTML={
          {
            "__html": {},
          }
        }
      />
      <div
        className="css-yl0smr-Heading"
      >
        header__app_title
      </div>
    </div>
    <div
      className="css-1vepu24-AppHeader"
    >
      <span
        className="css-1kymlcf-AppHeader"
      >
        project__end_successful_header
      </span>
    </div>
    <div
      className="css-avzpqv-ProgressTimeClock"
      onClick={[Function]}
    >
      <div
        className="css-1xtqo1u-Tooltip"
        id="tooltip_42"
      >
        <div
          className="css-1x45ysw-ProgressTimeClock"
        >
          <div
            className="css-fmthw-Text"
          >
            01.01.1970 - 00:00
          </div>
          <div
            className="css-kgdmdb-Icon"
            dangerouslySetInnerHTML={
              {
                "__html": {},
              }
            }
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="css-1hykuu4-DesktopBackgroundOverlay"
  >
    <div
      className="css-frr1bk-DesktopBackgroundOverlay"
    >
      <div
        className="desktop-final-content css-17kczyk-ProjectEnd"
      >
        <div
          className="css-1spg3d3-Heading"
        >
          project__end_successful_title
        </div>
        <div
          className="css-fmthw-Text"
        >
          project__end_successful_text
        </div>
        <div
          className="css-wx6fpr-ProjectEnd"
        >
          <div
            className="css-8050ac-Heading"
          >
            project__start_dialog_text_confirm_modal_code
          </div>
          <div
            className="css-ke0ydv-Text"
          >
            fsdskdjf
          </div>
        </div>
      </div>
    </div>
    <div
      className="css-6ndbya-DesktopBackground"
    >
      <div
        className="background-logo css-1xk0bda-DesktopBackground"
        dangerouslySetInnerHTML={
          {
            "__html": {},
          }
        }
      />
    </div>
  </div>
</div>
`;
